src/sentry/api/endpoints/group_events.py

```diff
@@ -40,16 +41,20 @@ def get(self, request, group):
         )
 
         query = request.GET.get('query')
+
         if query:
             try:
                 query_kwargs = parse_query(group.project, query, request.user)
             except InvalidQuery as exc:
                 return Response({'detail': six.text_type(exc)}, status=400)
 
             if query_kwargs['query']:
-                events = events.filter(
-                    message__icontains=query_kwargs['query'],
-                )
+                q = Q(message__icontains=query_kwargs['query'])
+
+                if len(query) == 32:
+                    q |= Q(event_id__exact=query_kwargs['query'])
```

Comment:
This is a nit, but `__exact` is implied. So you can reduce this down to just `q |= Q(event_id=query_kwargs['query'])`